### PR TITLE
Removes the non-functioning Mesh shape from the code base.

### PR DESCRIPTION
**Pull request by moorepants**
Fixes issue #21.

This is a new PR that subsumes PR #158.

- [x] There are no merge conflicts.
- [x] If there is a related issue, a reference to that issue is in the
  commit message.
- [x] Unit tests have been added for the bug. (Please reference the issue #
  in the unit test.)
- [x] The tests pass both locally (run `nosetests`) and on Travis CI.
- [x] The code follows PEP8 guidelines. (use a linter, e.g.
  [pylint](http://www.pylint.org), to check your code)
- [x] The bug fix is documented in the [Release
  Notes](https://github.com/pydy/pydy#release-notes).
- [x] The code is backwards compatible. (All public methods/classes must
  follow deprecation cycles.)
- [x] Create a WIP PR with the mesh code.
- [x] All reviewer comments have been addressed.

### CODE REVIEW

**Comment by moorepants** [comment]
@pydy/pydy-developers This needs a quick review and merge.

**Comment by moorepants** [comment]
Note that I am not worrying about backwards incompatibility here because it never worked in the first place, so no one probably used it.

File: pydy/viz/tests/test_shapes.py [hunk 1]
```diff
@@ -253,56 +253,6 @@ def test_circle():
...
-    assert isinstance(mesh_shape, Shape)
-
-    mesh_shape_ = Mesh(points=point_list, color='green')
-    assert mesh_shape_.name == 'unnamed'
-    assert mesh_shape_.__str__() == \
-        'Mesh unnamed color:green material:default points:[[ 2.  3.  1.]\n [ 4.  6.  2.]\n [ 5.  3.  1.]\n [ 5.  3.  6.]\n [ 2.  8.  4.]\n [ 7.  4.  1.]]'
-    assert mesh_shape_.__repr__() == 'Mesh'
-
-
 def test_plane():
```
Comment (jcrist) [inline pydy/viz/tests/test_shapes.py]: Super minor, but these functions will have only 1 blank line between them after this is gone.

**Comment by jcrist** [comment]
Looks good - feel free to ignore the pep8 nitpick if you don't want to change it.

**Comment by moorepants** [comment]
@jcrist Can you check off the checklist and merge if ready?

**Pull request merged**